extensions/libraries/joomla_entity/src/Fields/Traits/HasFields.php

```diff
@@ -21,11 +22,25 @@
 trait HasFields
 {
 	/**
-	 * Associated fields.
+	 * Associated fields
 	 *
-	 * @var  Collection
+	 * @var  array
 	 */
-	protected $fields;
+	protected $fields = null;
+
+	/**
+	 * Flag to know if the cached fields have values attached or not
+	 *
+	 * @var  boolean
+	 */
+	protected $withValues = false;
+
+	/**
+	 * Field values
+	 *
+	 * @var  array
+	 */
+	protected $fieldValues = null;
```

Comment:
You don't need to explicitly initialise properties as null because they are null.